idb/postgres/postgres.go

```diff
@@ -708,9 +714,32 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 		// this should explicitly match the primary key on txn (round,intra)
 		query += " ORDER BY t.round, t.intra"
 	}
-	if tf.Limit != 0 {
-		query += fmt.Sprintf(" LIMIT %d", tf.Limit)
+
+	// Determine the LIMIT clause
```

Comment:
Are you actually seeing better performance with the group id filter overriding the limit filter (vs a straight limit)?

Reply:
Saw it on a mainnet database. If I remember correctly, the server was running pg14.6. @urtho can confirm.

@urtho came up with the idea of removing the `LIMIT`, and it did lead to a better execution plan.

I still don't understand why that happens though - in theory the `LIMIT` clause should not affect the execution plan.